### PR TITLE
[Mobile routes] Make result panel resizable

### PR DESCRIPTION
**Pull request by bbecquet**
## Description
Add resizable behavior to the route result panel on mobile.
The biggest part of this PR is actually simplifying the `<Panel>` component lifecycle for the management of the resize events, as well as the corresponding CSS.
Hopefully this makes it easier to read, but more importantly this fixes some UX problems with mobile panels, like the handler "jumping" below the cursor/finger and animations feeling awkward.

This is hard to validate with automatic testing, so if reviewers could play a bit with it on mobile devices that would be great :)

### CODE REVIEW

File: src/components/ui/Panel.jsx [hunk 1]
```diff
@@ -91,41 +94,41 @@ export default class Panel extends React.Component {
...
-    clearTimeout(this.timer);
 
-    const nativeEvent = event.nativeEvent;
-    const clientY = nativeEvent.changedTouches
-      ? nativeEvent.changedTouches[0].clientY
-      : nativeEvent.clientY;
-    const sizes = [0, window.innerHeight / 2, window.innerHeight];
-    const positionIndex = getClosestIndex(sizes, clientY);
+    document.removeEventListener('mousemove', this.moveCallback);
+    document.removeEventListener('touchmove', this.moveCallback);
```
Comment (G-Ray) [inline src/components/ui/Panel.jsx]: Should we also remove the listeners in componenWillUnmount, just in case ?

Comment (bbecquet) [reply src/components/ui/Panel.jsx]: Good idea, I'll add that. Thanks!

File: src/components/ui/Panel.jsx [hunk 2]
```diff
@@ -91,41 +96,41 @@ export default class Panel extends React.Component {
...
-    } else if (positionIndex === 0) {
-      size = 'maximized';
+    if (event.timeStamp - this.interactionStarted < CLICK_RESIZE_TIMEOUT_MS) {
+      // the resize handler was only clicked, provide 'smart' resize
+      size = !previousSize ? 'minimized' : null;
+    } else {
+      // the resize handler was really dragged-n-dropped
+      const sizes = [0, window.innerHeight / 2, window.innerHeight];
+      const positionIndex = getClosestIndex(sizes, currentHeight);
+      if (positionIndex === 0) {
```
Comment (G-Ray) [inline src/components/ui/Panel.jsx]: What about positionIndex === 1 ?

File: src/components/ui/Panel.jsx [hunk 3]
```diff
@@ -91,41 +96,41 @@ export default class Panel extends React.Component {
...
+      }
     }
 
-    this.holding = false;
     this.setState({
-      isTransitioning: true,
+      holding: false,
       size,
       currentHeight: null,
+      isTransitioning: true,
```
Comment (G-Ray) [inline src/components/ui/Panel.jsx]: I thing `isTransitioning` is not used anymore ?

File: src/components/ui/Panel.jsx (hunk 2 above)
Comment (bbecquet) [reply src/components/ui/Panel.jsx]: It's the medium position, i.e. the default case, which would result in `size=null`.

File: src/components/ui/Panel.jsx (hunk 3 above)
Comment (bbecquet) [reply src/components/ui/Panel.jsx]: It is for now. We need to call `updateMobileMapUI` after a CSS transition is done on the panel. It's currently done through `componentDidUpdate` when this state value is changed (see the lines below).
But you're right to spot it. It's an old hack with ugly coupling between CSS and JS… I'll try find an alternative using https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/transitionend_event

File: src/components/ui/Panel.jsx (hunk 2 above)
Comment (G-Ray) [reply src/components/ui/Panel.jsx]: ok, shouln't we asign `size = null` in this case ?

Comment (bbecquet) [reply src/components/ui/Panel.jsx]: It's already the initial value, line 105.

**Pull request merged**